standards/SPIFFE_Workload_API.md

```diff
@@ -0,0 +1,170 @@
+
+For additional clarity, please see [Appendix A](#appendix-a.-sample-implementation-state-machines) for sample implementation state machines.
+
+### 5.2 Federated Bundles
+
+The X.509-SVID Profile will always provide a Trust Bundle for the Trust Domain in which an SVID resides, however, it may also provide bundles for foreign Trust Domains. This enables workloads to communicate *across* Trust Domains, and is the primary mechanism through which federation is enabled. A bundle representing a foreign Trust Domain is known as a *Federated Bundle*.
+
+When authenticating a client from a foreign trust domain, the authenticator chooses the bundle representing the client’s presented trust domain for validation. Similarly, when authenticating a server, the client uses the bundle representing the server’s trust domain. If no matching bundle is present for the SVID in use, then the peer is untrusted. This approach is required in order to account for the lack of widespread support for SAN URI Name Constraints in common X.509 libraries. Please see [Section 4.2](X509-SVID.md#42-name-constraints) of the X509-SVID specification for more information.
+
+### 5.3 Default Identity
+
+It is often the case that a workload doesn’t know what identity it should assume. Determining when to assume what identity is a site-specific concern, and as a result, the SPIFFE specifications don’t reason about how to do this.
+
+In order to support the widest variety of use cases, the X.509-SVID Profile supports the issuance of multiple identities, while also defining a default identity. It is expected that workloads which are aware of multiple identities can handle decision making on their own. Workloads which don’t understand how to leverage multiple identities may use the default identity. The default identity is the first in the list. Protocol buffers ensure that the order of the list is preserved.
+
+### 5.4 Profile Messages
+
+The X.509-SVID Profile messages are expressed as a Protocol Buffer version 3 (proto3). They are defined below. For the specific service definition for this profile, please see the SPIFFE Workload API [Service Definition](#3-service-definition).
+
+```protobuf
+// The X509SVIDResponse message carries a set of X.509 SVIDs and their
+// associated information. It also carries a set of global CRLs, and a
+// TTL to inform the workload when it should check back next.
+message X509SVIDResponse {
+    // A list of X509SVID messages, each of which includes a single
+    // SPIFFE Verifiable Identity Document, along with its private key
+    // and bundle.
+    repeated X509SVID svids = 1;
+
+    // ASN.1 DER encoded
+    repeated bytes crl = 2;
```

Comment:
Are the CRLs here issued by CAs in this SVID's same trust domain? Any trust domain? Or just CRLs covering this SVID (up the cert chain to this SVID's root)?

Reply:
I think the intention here is to deliver CRLs for any/all trust domains...

Another way could be to ship them as a map, keyed by the trust domain they belong to... though, there may be more than one CRL per trust domain.

I should note that we discussed the possibility that the CRL signatures need not be checked by the client when being received over the workload API because the workload API is already trusted to deliver identity and trust roots. As a side-effect of that, the door would be open for a workload's trust domain to ship CRLs for foreign issuers. Whether this is a good idea or not, I'm not sure :).

Reply:
Having a map from trust domain to CRL(s) would probably help out client implementors.

As for not checking the signatures of the CRLs: sounds like there would be an implicit contract that server is expected to validate signatures as needed for all of its workloads. Expecting clients not to validate signatures sounds suspect to me... but if that's the intention here, would recommend documenting it.

Reply:
I opened https://github.com/spiffe/spiffe/issues/82 to track this